arch/arm/cpu/armv8/sysinfo.c

```diff
@@ -0,0 +1,55 @@
+#include <smbios.h>
+// #include <sysinfo.h>
+#include <stdio.h>
+
+void sysinfo_get_cache_info(u8 level, struct smbios_type7 *cache_info) {
+	unsigned long clidr_el1, ccsidr_el1;
+	unsigned csselr_el1;
+	printf("=====================> arm64 sysinfo_get_cache_info\n");
+
+	// Read CLIDR_EL1
+	asm volatile("mrs %0, clidr_el1" : "=r" (clidr_el1));
+	printf("CLIDR_EL1: 0x%lx\n", clidr_el1);
+
+	int cache_type = (clidr_el1 >> (3 * level)) & 0x7;
+	if (!cache_type)
+		return;  // No more cache levels
+
+	// Select cache level
+	csselr_el1 = (level << 1);  // Select data/unified cache (bit 0 = 0) and level
+	asm volatile("msr csselr_el1, %0" : : "r" (csselr_el1));
```

Comment:
There's ann 8.3 extension for ARMv8.3-CCIDX which provides extra 32 bits on top of the 64 for armv7 compat. 
We don't currently support 8.3 but I think we should add some provisioning for it

Reply:
I can add a note here and mask the higher 32 bits.